interactors/nonceTransactionsHandler.go

```diff
@@ -0,0 +1,135 @@
+package interactors
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+// nonceTransactionsHandler is the handler used for an unlimited number of addresses.
+// It basically contains a map of addressNonceHandler, creating new entries on the first
+// access of a provided address. This struct delegates all the operations on the right
+// instance of addressNonceHandler. It also starts a go routine that will periodically
+// try to resend "stuck transactions" and to clean the inner state. The recommended resend
+// interval is 1 minute. The Close method should be called whenever the current instance of
+// nonceTransactionsHandler should be terminated an collected by the GC.
+// This struct is concurrent safe.
+type nonceTransactionsHandler struct {
+	proxy       Proxy
+	mutHandlers sync.Mutex
+	handlers    map[string]*addressNonceHandler
+	cancelFunc  func()
+}
+
+func NewNonceTransactionHandler(proxy Proxy, intervalToResend time.Duration) (*nonceTransactionsHandler, error) {
```

Comment:
Not sure if a sdk user will now what to fill, here.

If the sdk controls this interval it can even make it dynamic, meaning it could try to send the tx's faster and if they still fail increase the timeout until it decides to do a permanent failure.